_episodes/04-use-license.md

```diff
@@ -5,7 +5,185 @@ exercises: 0
+The way to do it, is by assigning an open source licence to the code that you share with others, even if you are not publishing it but simply sharing it, e.g., by email.  
+Notice that "if you do not assign an open source license to your project, everybody who contributes to it becomes an exclusive copyright holder of their work. That means 'nobody' can use, copy, distribute, or modify their contributions – includes yourself." [[ref]](https://opensource.guide/legal/)   
+
+Also, you should be aware of whether other people's code, that you want to reuse or include in your project, has a licence and what type of licence it has. If you use, modify, or distribute a non licenced code, you are potentially liable to be sued by the code's owner.
+
+How can you know which permissions you have from a licence? How can you know whether a project has an open source licence? See [https://choosealicense.com/](https://choosealicense.com/)
+
+> ### Challenge
+>
+> Work in pairs. Ask learners to say wich permissions they have from a licence of a given software. Write permissions on sticky notes and stick them under each software name. Instructors should push learners to be more creative in terms of permissions. For example, can you sell the software? Can you sell support?
+{: .challenge}
+
+### Different licenses for different types of products (even data)
+
+
+## Tell why is important that a product/code has a licence
+
+* Why you need a licence
+* Explain the differences between concepts such as [open](https://opensource.org/docs/osd) and proprietary
+
+## Tell what is the importance of third-party dependencies on your product/code
+
+Third-party dependencies are any library that you are using in your project that was wrote by someone else. For example, if your data analysis in R uses ggplot2 for the visualisation, ggplot2 would be a third-party dependence. Third-party dependencies have their own licenses that can be different from the license that you apply to your project but they should be compatible. The freedom provided by your license must not violate the restriction described in the license of the third-party depencency that you are using. For example:
+
+The [GNU Affero General Public License](https://www.gnu.org/licenses/agpl-3.0.en.html) (AGPL) license says
+
+> if you modify the Program, your modified version must prominently offer all users interacting with it remotely through a computer network (if your version supports such interaction) an opportunity to receive the Corresponding Source of your version by providing access to the Corresponding Source from a network server at no charge, through some standard or customary means of facilitating copying of software.
+
+Therefore, if one of third-party libraries that you are using on your web app is licensed under this license, you should be aware that you will need to provide a copy of this AGPL third-party library that you are using.
```

Comment:
licenced? licence?

Reply:
And not just of the library, but also the rest of the code ("Corresponding Source'). It's a strong copyleft, your users are interacting with the whole thing (similarly to a statically linked binary), and that's what section 13 says.